src/lib.rs

```diff
@@ -454,6 +454,26 @@ enum Repr {
     },
 }
 
+impl PartialEq for Repr {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Heap(l0), Self::Heap(r0)) => Arc::ptr_eq(l0, r0),
+            (Self::Static(l0), Self::Static(r0)) => core::ptr::eq(l0, r0),
+            (
+                Self::Inline {
+                    len: l_len,
+                    buf: l_buf,
+                },
+                Self::Inline {
+                    len: r_len,
+                    buf: r_buf,
+                },
+            ) => l_len == r_len && l_buf == r_buf,
+            _ => false,
+        }
+    }
+}
```

Comment:
I'd rather have this be a method called `ptr_eq` or similar. Using `PartialEq` feels a bit wrong to me and yields no benefits over a simple private method as this is only used once for the `PartialEq<SmolStr> for SmolStr` impl